phare/pp/diagnostic/data/periodic_overlap.py

```diff
@@ -0,0 +1,42 @@
+def intralevel(clazz, diag, patches):
+    if len(patches) == 0:
+        return {}
+    if diag.dim == 1:
+        fn = _border_patches_1d
+
+    if fn is None:
+        raise ValueError(
+            "Periodic Overlap dimension unsupported, dimension: " + diag.dim
+        )
+    return {
+        dataset: fn(diag, patches, patches[0].patch_data.nGhosts(dataset))
+        for dataset in patches[0].patch_data.keys()
+    }
+
+
+def _border_patches_1d(diag, patches, nGhosts):
+    assert len(patches)
+
+    direction = "x"
+    max_domain = diag.sim.simulation_domain()
+    max_x = max_domain[0]
+    min_x = diag.sim.origin[0]
+
+    ghost_width = patches[0].patch_level.cell_width(direction) * nGhosts
```

Comment:
when you call the function in overlap.py

``` 
for dataset_key, border_patches in periodic_intralevel(
                clazz, diag, patch_level.patches
```

could you pass `patch_level` rather than patches

so 

```
    ghost_width = patches[0].patch_level.cell_width(direction) * nGhosts
```

in _border_patches_1d would just be

```
ghost_width =patch_level.cell_width(direction) * nGhosts
```

also 

```
    return {
        dataset: fn(diag, patches, patches[0].patch_data.nGhosts(dataset))
        for dataset in patches[0].patch_data.keys()
    }
```

could be

```
    return {
        dataset: fn(diag, patch_level, nGhosts(data_name))
        for data_name in patch_level.data_names()
    }
```

`patch_level.data_names()` could return the list of the names of all data stored in its patches patch datas no ? I guess all patches of a patch_level have the same patch_datas (plural) no ?

Reply:
patch_level could forward the dataset names sure

I have already refactored this to use patch_level in place of patches

Reply:
if you could expand on the

"patches[0].patch_data.nGhosts(dataset)" -> "nGhosts(data_name)" 

part please that would be appreciated.

Reply:
I don't have impl details but all I'm saying is that in principle the number of ghosts only depend on the final quantity : Bx or Bz etc. so I don't see that nGhost() should be a patch_data method.

Reply:
I've done something similar to "patch_level.data_names()"